unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -3054,6 +3054,12 @@ bool includeInvisible
             AgentHand.transform.rotation = this.transform.rotation;
         }
 
+        // set random seed used by unity
+        public void SetRandomSeed(int seed) {
+            UnityEngine.Random.InitState(seed);
+            actionFinishedEmit(true);
+        }
```

Comment:
What about the `System.Random` seed?

Reply:
I've added a new `systemRandom` variable to BaseFPSAgentController, which is now used in place of always constructing a new `System.Random()` object.

Some notes:
1. There was no way to set the seed of `System.Random` globally.
2. `System.Random` is necessary for generating doubles, instead of just floats (which can be done with `UnityEngine.Random`).
3. It is intentionally not used with a few actions that have `randomSeed` as a parameter. The only public one is InitialRandomSpawn, which is being reworked, such that, among other things, it will no longer have a random seed parameter.
4. After calling `controller.reset()`, there are no guarantees that the seeds remain in tact.